File: frontend/src/features/subscriptions/api/SubscriptionService.ts
import { DefaultApi } from '../../../api-client';
import { apiClient } from '../../../utils/api';

const api = new DefaultApi(undefined, '', apiClient);

export const SubscriptionService = {
  getAll: async (workspaceId: string) => {
    const response = await api.getSubscriptions(workspaceId);
    return response.data;
  },
  create: async (workspaceId: string, s: any) => { 
    const response = await api.createSubscription(workspaceId, s);
    return response.data;
  },
  update: async (workspaceId: string, sId: string, s: any) => { 
    const response = await api.updateSubscription(workspaceId, sId, s);
    return response.data;
  },
  delete: async (workspaceId: string, sId: string) => {
    await api.deleteSubscription(workspaceId, sId);
  },
};

File: backend/src/main/java/com/keeper/subscription/SubscriptionService.java
package com.keeper.subscription;

import com.keeper.dto.*;
import java.util.*;

public interface SubscriptionService {
  SubscriptionDTO createSubscription(UUID workspaceId, SubscriptionDTO subscriptionDTO);

  void deleteSubscription(UUID workspaceId, UUID id);

  java.util.List<SubscriptionDTO> getSubscriptions(UUID workspaceId);

  SubscriptionDTO updateSubscription(UUID workspaceId, UUID id, SubscriptionDTO subscriptionDTO);
}
